scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/shared/SchedulerConfig.java

```diff
@@ -135,15 +129,53 @@ public String getRestUrl() {
         return properties.get(REST_URL);
     }
 
+    /**
+     * @return the URL from the window location
+     */
+    public String getWindowsLocationUrl() {
+        String urlFromCurrentLocation = com.google.gwt.user.client.Window.Location.getHref();
+        urlFromCurrentLocation = urlFromCurrentLocation.replace(com.google.gwt.user.client.Window.Location.getPath(), "");
+        return urlFromCurrentLocation;
+    }
+
+    /**
+     * @return the REST_PUBLIC_URL if it is set or take it from the window location
+     */
     @Override
-    public String getRestPublicUrlIfDefinedOrOverridden() { return properties.get(REST_PUBLIC_URL);}
+    public String getRestPublicUrlIfDefinedOrOverridden() {
+        String restPublicUrl = properties.get(REST_PUBLIC_URL);
+        if (restPublicUrl == null) {
+            String restUrlFromCurrentLocation = getWindowsLocationUrl();
+            restUrlFromCurrentLocation += "/rest";
+            return restUrlFromCurrentLocation;
+        }
+        return restPublicUrl;
+    }
 
+    /**
+     * @return the NOVNC_URL if it is set or take it from the window location
+     */
     public String getNoVncUrl() {
-        return properties.get(NOVNC_URL);
+        String noVncUrl = properties.get(NOVNC_URL);
+        if (noVncUrl == null) {
+            String noVncUrlFromCurrentLocation = getWindowsLocationUrl();
+            noVncUrlFromCurrentLocation += "/rest/novnc";
+            return noVncUrlFromCurrentLocation;
+        }
+        return noVncUrl;
     }
 
+    /**
+     * @return the NOVNC_PAGE_URL if it is set or take it from the window location
+     */
     public String getNoVncPageUrl() {
-        return properties.get(NOVNC_PAGE_URL);
+        String noVncPageUrl = properties.get(NOVNC_PAGE_URL);
+        if (noVncPageUrl == null) {
```

Comment:
additionally (and this applies for both VNC urls), the colon character must be escaped, as the url will be passed to another url as a parameter